src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -1174,6 +1253,35 @@ LRESULT CALLBACK WindowImplWin32::globalOnEvent(HWND handle, UINT message, WPARA
     return DefWindowProcW(handle, message, wParam, lParam);
 }
 
+////////////////////////////////////////////////////////////
+void WindowImplWin32::prepareTouch()
+{
+    static bool prepared = false;
+    if (!prepared)
+    {
+        prepared = true;
+
+        // Reset touch IDs
+        for (int i = 0; i < 10; ++i)
+            touchIDs[i] = static_cast<DWORD>(-1);
+    }
+
+    if (m_handle)
+        RegisterTouchWindow(m_handle, 0);
+}
+
+////////////////////////////////////////////////////////////
+bool WindowImplWin32::isTouchDown(unsigned int finger)
+{
+    return touchIDs[finger] != static_cast<DWORD>(-1);
+}
+
+////////////////////////////////////////////////////////////
+Vector2i WindowImplWin32::getTouchPosition(unsigned int finger)
+{
+    return Vector2i(touches[finger].x, touches[finger].y);
```

Comment:
Maybe
```cpp
assert(isTouchDown(finger));
```
just to avoid logic errors.

Reply:
Here I'm not 100% sure (but I'm also not sure how other platforms handle this without looking into it right now). If the touch stops, the position isn't reset, so it would still return the last known position.

Reply:
I see. Maybe we should define the behavior here, document it and enforce it in code.

The user will likely have the same questions as we do 🤔